etcdserver/config.go

```diff
@@ -61,6 +61,12 @@ type ServerConfig struct {
 	ClientCertAuthEnabled bool
 
 	AuthToken string
+
+	// OnShutdown gracefully stops gRPC server on shutdown.
```

Comment:
this should be a generic thing instead of talking about grpc, etc

`// OnShutdown is called immediately before releasing etcd server resources.`